app/services/booking_responder.rb

```diff
@@ -1,16 +1,20 @@
 class BookingResponder
   delegate :visit, to: :staff_response
 
-  def initialize(staff_response, message = nil)
-    @staff_response = staff_response
-    @message = message
+  def initialize(visit:, user: nil, message: nil)
```

Comment:
They work well here because both user and message are optional, so you get cases where you pass only the user, or only the message, none, or both.